lib/mesh.js

```diff
@@ -1700,17 +1700,18 @@ Mesh.prototype._integrateComponent = function(spec, done) {
 
   componentInstance = _this._mesh.elements[componentName].component.instance;
   exchange = componentInstance.exchange;
-  event = componentInstance.event;
-  localEvent = componentInstance.localEvent;
+  event = Internals._decoupleEventAPILayer(componentInstance.event);
+  localEvent = Internals._decoupleEventAPILayer(componentInstance.localEvent);
   localEventEmitter = componentInstance.localEventEmitter;
 
   Object.keys(this.exchange).forEach(function(otherComponentName) {
     exchange[otherComponentName] = _this.exchange[otherComponentName];
```

Comment:
Would suggest changing this to
if (otherComponentName === "$call") return;
event... = _this.event...
(if otherComponentName === this._mesh...) return;
localEvent... = _this.localEvent...